src/native/libs/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -10,31 +10,55 @@
 
 #define SUCCESS 1
 
-static const EVP_MD* g_evpFetchMd5 = NULL;
-static pthread_once_t g_evpFetch = PTHREAD_ONCE_INIT;
-
-static void EnsureFetchEvpMdAlgorithms(void)
-{
-    // This is called from a pthread_once - this method should not be called directly.
-
 #ifdef NEED_OPENSSL_3_0
-    if (API_EXISTS(EVP_MD_fetch))
-    {
-        ERR_clear_error();
-
-        // Try to fetch an MD5 implementation that will work regardless if
-        // FIPS is enforced or not.
-        g_evpFetchMd5 = EVP_MD_fetch(NULL, "MD5", "-fips");
+#define SETUP_MD_FETCH(export, fn, name, query) \
```

Comment:
The main reason for macroizing this is so that each algorithm is pay to play. That is, if I want to SHA-2-256, I shouldn't have to pay the price for an EVP_fetch for SHA-1. So every hash algorithm is its own lazy PTHREAD_ONCE_INIT. If we want, we can combine everything in to one init. The benefit for that is if can probably avoid this... nice... macro, the downside being the very first call to any hash function is going to initialize all of them at once.